plugins/base/src/main/kotlin/transformers/documentables/DocumentableFilter.kt

```diff
@@ -323,5 +339,263 @@ internal class DocumentableVisibilityFilter(val context: DokkaContext) : PreMerg
+        ): Pair<Boolean, List<DProperty>> = properties.filter {
+            it.isAllowedInPackage()
+        }.let {
+            Pair(properties.size != it.size, it)
+        }
+
+        private fun filterEnumEntries(entries: List<DEnumEntry>) =
+            entries.filter { it.isAllowedInPackage() }.map { entry ->
+                DEnumEntry(
+                    entry.dri,
+                    entry.name,
+                    entry.documentation,
+                    entry.expectPresentInSet,
+                    filterFunctions(entry.functions).second,
+                    filterProperties(entry.properties).second,
+                    filterClasslikes(entry.classlikes).second,
+                    entry.sourceSets,
+                    entry.extra
+                )
+            }
+
+        private fun filterClasslikes(
+            classlikeList: List<DClasslike>
+        ): Pair<Boolean, List<DClasslike>> {
+            var modified = false
+            return classlikeList.filter {
+                when (it) {
+                    is DClass -> it.isAllowedInPackage()
+                    is DInterface -> it.isAllowedInPackage()
+                    is DEnum -> it.isAllowedInPackage()
+                    is DObject -> it.isAllowedInPackage()
+                    is DAnnotation -> it.isAllowedInPackage()
+                }
+            }.map {
+                fun helper(): DClasslike = when (it) {
+                    is DClass -> DClass(
+                        it.dri,
+                        it.name,
+                        filterFunctions(it.constructors).let {
+                            modified = modified || it.first; it.second
+                        },
+                        filterFunctions(it.functions).let {
+                            modified = modified || it.first; it.second
+                        },
+                        filterProperties(it.properties).let {
+                            modified = modified || it.first; it.second
+                        },
+                        filterClasslikes(it.classlikes).let {
+                            modified = modified || it.first; it.second
+                        },
+                        it.sources,
+                        it.visibility,
+                        it.companion,
+                        it.generics,
+                        it.supertypes,
+                        it.documentation,
+                        it.expectPresentInSet,
+                        it.modifier,
+                        it.sourceSets,
+                        it.extra
+                    )
+                    is DAnnotation -> DAnnotation(
+                        it.name,
+                        it.dri,
+                        it.documentation,
+                        it.expectPresentInSet,
+                        it.sources,
+                        filterFunctions(it.functions).let {
+                            modified = modified || it.first; it.second
```

Comment:
I think there are way to many "it-scopes" open and things should have names at this point 🙄